core/src/rpc_service.rs

```diff
@@ -260,20 +263,37 @@ impl JsonRpcService {
             .build()
             .expect("Runtime");
 
-        let bigtable_ledger_storage = if config.enable_bigtable_ledger_storage {
-            runtime
-                .block_on(solana_storage_bigtable::LedgerStorage::new(false))
-                .map(|x| {
-                    info!("BigTable ledger storage initialized");
-                    Some(x)
-                })
-                .unwrap_or_else(|err| {
-                    error!("Failed to initialize BigTable ledger storage: {:?}", err);
-                    None
-                })
-        } else {
-            None
-        };
+        let exit_bigtable_ledger_upload_service = Arc::new(AtomicBool::new(false));
+
+        let (bigtable_ledger_storage, _bigtable_ledger_upload_service) =
```

Comment:
It's slightly awkward that `BigTableUploadService` is plumbed into `RpcService` like this.   Moving it up a level into `Validator` would require moving the tokio Runtime and `solana_storage_bigtable::LedgerStorage::new` as well and I didn't like that code churn in this PR.    That's my excuse at least